modules/transport-netty4/src/main/java/org/elasticsearch/http/netty4/Netty4HttpPipeliningHandler.java

```diff
@@ -181,6 +209,27 @@ private void doWrite(ChannelHandlerContext ctx, Netty4HttpResponse readyResponse
         writeSequence++;
     }
 
+    private void doWrite(ChannelHandlerContext ctx, Netty4ChunkedHttpResponse readyResponse, ChannelPromise promise) throws IOException {
+        assert currentChunkedWrite == null : "unexpected existing write [" + currentChunkedWrite + "]";
+        final PromiseCombiner combiner = new PromiseCombiner(ctx.executor());
+        currentChunkedWrite = new ChunkedWrite(combiner, promise, readyResponse);
+        final ChannelPromise first = ctx.newPromise();
+        combiner.add((Future<Void>) first);
+        enqueueWrite(ctx, readyResponse, first);
+        while (ctx.channel().isWritable()) {
+            if (writeChunk(ctx, combiner, readyResponse.body())) {
```

Comment:
This looks slightly odd to me, might be me not fully understanding if a write can happen between the channel becoming writable and the `channelWritabilityChanged` method being called. But if `enqueueWrite` ever enqueues the `readyResponse`, it seems wrong to start writing the chunk now? I suppose you assume that `queuedWrites` is empty, but I am not sure that is always the case. If a split write happens first and then a chunked write (due to pipelining), would we then not risk the chunk being written here first?

Reply:
:facepalm: again, you're right this might be an issue we could run into. I pushed https://github.com/elastic/elasticsearch/pull/88311/commits/ee09bb92b7e185130d2050bb55187d1de04a0f5e to as you point out, cover the case where we only queue the first chunk because there's a pending split write of a huge non-chunked message.